### PR TITLE
Added tests for error_page_debug_messages

### LINKED ISSUE #232: Add Tests for error page debug configuration
Add tests for PR #227

### PR DESCRIPTION
**Pull request by cnovak**
Fixes #232

### CODE REVIEW

**Comment by cnovak** [comment]
New tests pass:
Screen Shot 2019-06-19 at 10 08 30 PM

File: tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Routing\Matcher\UrlMatcherInterface;
+use Symfony\Component\Routing\RequestContext;
+
+/**
+ * Test EdgeExceptionSubscriber.
+ *
+ * @group apigee_edge
+ */
+class EdgeExceptionSubscriberTest extends UnitTestCase {
```
Comment (mxr576) [inline tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php:43]: Why not `\Drupal\KernelTests\KernelTestBase`?

File: tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+  protected $exception;
+  protected $http_kernel;
+  protected $logger;
+  protected $redirect_destination;
+  protected $access_unaware_router;
+  protected $config_factory;
+  protected $messenger;
+  protected $get_response_for_exception_event;
```
Comment (mxr576) [inline tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php]: Property names should be camelCased.

File: tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    $request = $this->prophesize(RequestContext::class);
+    $this->access_unaware_router = $this->prophesize(UrlMatcherInterface::class);
+    $this->access_unaware_router->getContext(Argument::any())->willReturn($request->reveal());
+
+    $this->messenger = $this->prophesize(MessengerInterface::class);
+
+    $request = $this->prophesize(Request::class);
+
+    // Set empty objects so that clone on request object works.
+    $request->query = new stdClass();
```
Comment (mxr576) [inline tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php:75]: Well, these should be mocked too.

File: tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    $this->logger = $this->prophesize(LoggerInterface::class);
+    $this->redirect_destination = $this->prophesize(RedirectDestinationInterface::class);
+
+    $request = $this->prophesize(RequestContext::class);
+    $this->access_unaware_router = $this->prophesize(UrlMatcherInterface::class);
+    $this->access_unaware_router->getContext(Argument::any())->willReturn($request->reveal());
+
+    $this->messenger = $this->prophesize(MessengerInterface::class);
+
+    $request = $this->prophesize(Request::class);
```
Comment (mxr576) [inline tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php:72]: Maybe mocking with `$this->getMockbuilder(Request::class)->disableOriginalConstructor()->getMock()` would be easier.

**Review by mxr576 — CHANGES_REQUESTED** [review]
We already have a UI test that could also test this behavior which could ensure the error message is properly displayed on the UI.
`\Drupal\Tests\apigee_edge\Functional\ErrorHandlerTest`

File: tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php (hunk 2 above)
Comment (cnovak) [reply tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php]: Done

File: tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php (hunk 1 above)
Comment (cnovak) [reply tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php:43]: Created unit test to speed up testing.

File: tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php (hunk 3 above)
Comment (cnovak) [reply tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php:75]: Is there a reason you think they should be mocked?

File: tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php (hunk 1 above)
Comment (mxr576) [reply tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php:43]: Kernel tests are fast but require less additional work. But as I mentioned in the review comment, we already have a UI test that could be extended to validate this as well.

**Review by arlina-espinoza — APPROVED** [review]
Previously suggested changes are only for improvements and do not add to the logic being tested, this is good enough as it is. Thanks @cnovak :)

**Pull request merged**